admin/superuser-init.py

```diff
@@ -1,290 +1,139 @@
 #!/usr/bin/python
 import sys, os, shutil
 import json
+import re
 import argparse
 import getpass
 import openreview
-import utils
 import datetime
 
 """
-REQUIRED ARGUMENTS
+This script should be used by OpenReview administrators to automatically build
+the basic infrastructure for a new conference. It will create (or fill in as
+needed) a directory under openreview-scripts/venues with the necessary
+subdirectories, and will generate the scripts, webfields, and process functions
+needed to run a simple conference.
 
+REQUIRED KEYWORD ARGUMENTS
     conf - the full path of the conference group you would like to create.
         e.g. auai.org/UAI/2017
+        A directory with path exactly equal to the conference group ID
+        should already exist before running this script.
 
-OPTIONAL ARGUMENTS
+    data - a file containing parameters of the conference.
+        See openreview-scripts/admin/conference-template/params.data
 
+OPTIONAL KEYWORD ARGUMENTS
     overwrite - if present, overwrites the conference directory.
-    data - a .json file containing parameters that are otherwise prompted for in the script
     baseurl -  the URL of the OpenReview server to connect to (live site:
         https://openreview.net)
     username - the email address of the logging in user
     password - the user's password
 
-Gathers information to fill in config.py and the webfield/conf.html.
-Creates directories and groups for the entire CONF directory structure.
-Creates sub-directores python/, process/, data/ and webfield/.
-Creates config.py, admin-init.py, submissionProcess.js, commentProcess.js and conf.html.
-Optionally creates admin user profile
+
 """
 
+def parse_json(file):
+    json_string = open(os.path.join(os.path.dirname(__file__), file), 'r').read()
+    json_parsed = re.sub('\/\*[^*]+\*\/|\s|\n', '', json_string)
+    return json.loads(json_parsed)
+
+def build_directory(directory_path):
+
+    # create the subdirectories if they don't exist
+    for subpath in ['','/python','/webfield','/process','/data']:
+        path = '{0}/{1}'.format(directory_path, subpath)
+        if not os.path.exists(path):
+            print "Creating directory {0}".format(path)
+            os.makedirs(path)
+
+
+def generate_file(template_path, directory_path, data, overwrite = False):
+
+    # generate new files from templates
+    ext = '.py' if 'python' in template_path else '.js'
+    newfile_path = directory_path + '/' + template_path.replace('.template', ext)
+
+    with open(os.path.join(os.path.dirname(__file__), './conference-template/{0}'.format(template_path))) as template:
+        template_string = template.read()
+
+    for replacement in data:
+        template_string = template_string.replace(replacement, data[replacement])
+
+    if not os.path.exists(newfile_path) or overwrite:
```

Comment:
Move this check up to line 57. No reason to recreate the file if it's not going to be used.